test/meson.build

```diff
@@ -14,6 +14,14 @@ td += executable('readdir_inode', 'readdir_inode.c',
                  include_directories: include_dirs,
                  install: false)
 
+td += shared_library('test746', 'test746.c',
```

Comment:
@Nikratio Meson adds the default project flags as well, and I can't seem to figure out how to  prevent that without creating a new project and add a more complex configuration. There is an `override_options` parameter but it seems that it needs a newer Meson version to work. Any Meson expert around?

This builds a working library for now but I can't guarantee that it will not break in the future if the project flags change.

Reply:
> but it seems that it needs a newer Meson version to work

Can you describe what you mean by this? The kwarg itself was added in meson 0.40.0 which is significantly older than the minimum version required by libfuse (0.51.0).

Which default project flags are you concerned about, by the way? Stuff like `-O2 -g`? Why would this cause a problem here?

Reply:
### Meson
I saw that there is a way to override the default arguments using `override_options`.
Trying this with the `link_flags` key it gives this error:
```
WARNING: Project targets '>= 0.51' but uses feature introduced in '0.53.0': Dictionary entry using non literal key.
../test/meson.build:17:0: ERROR: Unknown variable "link_args".
```
Somewhere else I saw that the key should be as a string (e.g. `'link_args'`), but this still gives this error:
```
    if k.lang:
       ^^^^^^
AttributeError: 'str' object has no attribute 'lang'
../test/meson.build:17:0: ERROR: Unhandled python exception
    This is a Meson bug and should be reported!
```

As I mentioned I don't know much about Meson so I might be trying to do this entirely the wrong way.

### Compile flags
I add in the specific target the flags that are needed for the library's specific use, but this target configuration cannot guarantee that things won't change and break in the future. It's just a personal preference to keep such tests as tight as possible as if something changes and it breaks then the whole test could be invalidated and keep giving false positives and be very hard to notice.

The final combined compilation flags by Meson are:   
```
[1/2] cc -Itest/libtest746.so.p -Itest -I../test -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64
-Wall -Winvalid-pch -Wextra -O2 -g -D_REENTRANT -DHAVE_LIBFUSE_PRIVATE_CONFIG_
-Wno-sign-compare -Wstrict-prototypes -Wmissing-declarations -Wwrite-strings -fno-strict-aliasing
-Wno-unused-result -DHAVE_SYMVER_ATTRIBUTE -fPIC -MD -MQ test/libtest746.so.p/test746.c.o -MF
test/libtest746.so.p/test746.c.o.d -o test/libtest746.so.p/test746.c.o -c ../test/test746.c

[2/2] cc  -o test/libtest746.so test/libtest746.so.p/test746.c.o -Wl,--as-needed -Wl,--no-undefined
-shared -fPIC -Wl,--start-group -Wl,-soname,libtest746.so -Wl,-init,test746_init -Wl,-z,initfirst
-ldl -Wl,--end-group
```

Reply:
What option are you trying to override? "link_args" isn't a builtin option: https://mesonbuild.com/Builtin-options.html#compiler-options

Reply:
> It's just a personal preference to keep such tests as tight as possible as if something changes and it breaks then the whole test could be invalidated and keep giving false positives and be very hard to notice.

I'm not sure I buy this logic.

You could say by the same token that you prefer to keep the project runtime "as tight as possible" because if something changes and the program breaks then nothing works anymore.

Reply:
This discussion nicely illustrates the complexity of adding a separate LD_PRELOAD for this test. Let's try the approach with a separate `tests/issue_746_helper.c` FUSE daemon first before we continue this discussion please.

Reply:
@eli-schwartz
It's the compiling and linker flags to ensure that the generated object code, the initialisation of the preloaded library, and the symbol overloading work properly. Initially I used `custom_target` to compile this with fixed compiler settings, but I was told in a previous review to use the default Meson library build functions.

@Nikratio
Ok I'll remove all the extra stuff and add a minimal clone of passthrough_fh as a helper fuse daemon for the test.
Ten developers would approach this in ten different ways. I would prefer the PRELOAD method as that opens up the option to expand this test and easily add stress-testing tests to all passthrough examples, with various fixed and random runtime delay/fault injections without the need to create and maintain individual helper clones.

Reply:
Changes included in next patch.